docs/recap/bitvm2_fast_recap.tex

```diff
@@ -0,0 +1,229 @@
+
+In the ever-evolving landscape of blockchain technology, the desire to
+execute increasingly complex and large-scale programs on the
+Bitcoin\autocite{bitcoin_paper} network is growing. However, Bitcoin
+Script, the native programming language of Bitcoin, imposes strict
+size limits, such as the 4-megabyte cap on transaction inputs, which
+makes it challenging to implement certain advanced cryptographic
+proofs, like Zero Knowledge proofs. To address this limitation, the
+BitVM2\autocite{bitvm2} proposal introduces an innovative approach
+that enables the optimistic execution of large programs on the Bitcoin
+chain. This method leverages fraud proofs to ensure the validity of
+executions, offering a robust mechanism for verifying computations,
+while providing a fail-safe against incorrect or malicious
+operations. This document provides a concise overview of BitVM2,
+highlighting its potential to unlock new possibilities for secure and
+scalable program execution on Bitcoin.
+
+\section{Program splitting}\label{sec:program-splitting}
+
+\begin{quote}
+  Let there be a large program \(f\), which can be described in Bitcoin
+  Script. We want to compute it \textbf{\textbf{on-chain}}, i.e., find
+  such an \(f(x) = y\). To achieve this, we divide the program into
+  \(n\) sub-programs \(f_1, \ldots, f_n\) and their corresponding
+  intermediate states \(z_1, \ldots, z_n\), such that:
+
+  \begin{equation}
+    \begin{aligned}
+      f_1(x) &= z_1 \\
+      f_2(z_1) &= z_2 \\
+      &\cdots \\
+      f_n(z_{n-1}) &= y
+    \end{aligned}
+  \end{equation}
+\end{quote}
+
+However, the user (referred in BitVM2 as the operator) only needs to
+prove that the given program \(f\) indeed returns \(y\) for \(x\), or
+\textbf{give others the opportunity} to \textbf{disprove this
+fact}. In our case, this means giving challengers the ability to
+prove that at least one of the sub-programs \(f_i(z_{i-1}) \neq z_i\).
+
+\section{\textbf{Assert} transaction}\label{sec:assert-tx}
+
+To achieve this, the user publishes an \textbf{Assert} transaction,
+which has one output with multiple possible spending scenarios:
+
+\begin{enumerate}
+  \item \textbf{PayoutScript} (\textbf{LockTime} + signature) --- the
+    transaction has passed verification, and the operator can spend the
+    output, thereby confirming the statement \(f(x) = y\).
+  \item \textbf{DisproveScript} --- one of the challengers has found a
+    discrepancy in the intermediate states \(z_i\), \(z_{i-1}\) and the
+    sub-program \(f_i\). In other words, they have proven that
+    \(f_i(z_{i-1}) \neq z_i\), and thus, they can spend the output.
+\end{enumerate}
+
+\subsection{\texttt{DispoveScript}}\label{sec:dispove-script}
+
+\texttt{DispoveScript} is part of the MAST tree in a Taproot address,
+which allows challengers to claim the transaction amount for states
+\(z_i\), \(z_{i-1}\), and sub-program \(f_i\), is called
+\(DisproveScript_i\) in BitVM2:
+
+\begin{verbatim}
```

Comment:
Later, put it in the `algorithm` package

Reply:
Do you want me to do this in this PR?